### PR TITLE
Lazerbeak12345 add tab panel aguments

### PR DESCRIPTION
**Pull request by Lazerbeak12345**
A while ago (in pr #1075)  I added the `on-reorder` and `on-close-request` functions to the `Tab-Panel%` type, but I at the time didn't know how `augment` worked so I didn't actually add them correctly. This should fix this problem and allow these events to be augmented.

### CODE REVIEW

**Comment by capfredf** [comment]
can you add a test for this?

**Comment by Lazerbeak12345** [comment]
Yeah sure. Where would that test go?

**Comment by capfredf** [comment]
you can put your test file under `typed-racket-test/succeed/`

**Comment by Lazerbeak12345** [comment]
I noticed that there is also `typed-racket-test/gui/succeed`. As this is a test of something in `typed/racket/gui` should this test I'll write actually go there?

**Comment by capfredf** [comment]
Ah, yes, that is indeed better

**Comment by Lazerbeak12345** [comment]
It's possible that I didn't write the test correctly, but I'm pretty sure that this should be accurate, or at least very close.

**Comment by capfredf** [comment]
LGTM

**Review by capfredf — APPROVED** [review]

**Comment by Lazerbeak12345** [comment]
Sorry about that wait, but I'm 100% sure if that test worked then we're good. :)

File: typed-racket-test/gui/succeed/test-tab-augments.rkt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      (on-close-request-callback index))))
+(define tabs : (Instance Tab-Panel-With-Callbacks%)
+  (new tab-panel-with-callbacks%
+       [choices '("one" "two" "three")]
+       [parent frame]
+       [style '(can-reorder can-close no-border)]
+       [on-reorder-callback (lambda (tab-list)
+                              (displayln (format "on-reorder ~v" tab-list)))]
+       [on-close-request-callback
+        (lambda (tabs)
```
Comment (Lazerbeak12345) [inline typed-racket-test/gui/succeed/test-tab-augments.rkt:25]: This is a bit of a nitpick as this variable name probably doesn't matter that much, but should I change this to something more correct, such as `index`?

Comment (capfredf) [reply typed-racket-test/gui/succeed/test-tab-augments.rkt:25]: Thanks, that's a good idea! I will change it in my next pull request.

Comment (capfredf) [reply typed-racket-test/gui/succeed/test-tab-augments.rkt:25]: Fixed it in #1113

**Pull request merged**